### PR TITLE
Add support for reading dataframe rownames

### PR DESCRIPTION
**Pull request by peiyaoli2**
Hi! I added a row name handler to recognize rownames in r dataframes (related to #12). This does not apply to matrices though.

### CODE REVIEW

File: src/rdata.h [hunk 1]
```diff
@@ -1,3 +1,9 @@
+#ifndef INCLUDE_RDATA_H
+#define INCLUDE_RDATA_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
```
Comment (evanmiller) [inline src/rdata.h]: Please split the #ifdef header changes into a separate pull request.

File: src/rdata.h [hunk 2]
```diff
@@ -80,6 +86,7 @@ typedef struct rdata_parser_s {
     rdata_table_handler         table_handler;
     rdata_column_handler        column_handler;
     rdata_column_name_handler   column_name_handler;
+	rdata_column_name_handler   row_name_handler;
```
Comment (evanmiller) [inline src/rdata.h]: Please fix indentation (four spaces instead of tabs)

**Comment by peiyaoli2** [comment]
It should be all fixed now! I will have another pull request for the header stuff. Do you think the MSVC support stuff (https://github.com/jonathon-love/librdata/commit/e85c23cc1fb95d5c9139a566b56331f3fae3c29a) can potentially be merged too?

**Comment by evanmiller** [comment]
Thanks. Yes feel free to open a PR with the MSVC changes.

**Pull request merged**